### ISSUE TITLE
build is not passing on Chineese (non English) locale

### ISSUE THREAD

**Issue by jokowncode** [checkstyle/checkstyle#7911]
If you locale is not English, then you may face some errors when maven verify ,the checkstyle validation cannot pass, like [#7644 error](https://github.com/checkstyle/checkstyle/issues/7644#issuecomment-600472495).

From [# 7644 comment](https://github.com/checkstyle/checkstyle/issues/7644#issuecomment-600646979), We know that checkstyle is strictly tested in the English environment, so the test will fail in the non-English environment.

So the problem is how to force the locale to be changed to English during build? (of course, you can also use the environment created by CI for build)

**Comment by jokowncode** [comment]
My solution is to add the parameter`-Duser.language=en -Duser.region=GB` when running maven clean verify.
For example, in IDEA, you can modify it in `File-> Setting-> Build,Execution,Deployment-> Build Tools-> Maven-> Runner-> VM options`.

**Comment by bmhm** [comment]
@jokowncode good catch. 
Files: https://github.com/checkstyle/checkstyle/blob/master/src/xdocs/beginning_development.xml

This will make a good first issue.

**Comment by jokowncode** [comment]
I am on it.

**Comment by jokowncode** [comment]
@bmhm When I tried again, I found that the following configuration had to be added to the pom.xml file of maven to take effect.
```
<configuration>
    <argLine>-Duser.language=en -Duser.region=GB</argLine>
</configuration>
```

**Comment by bmhm** [comment]
An alternative might be to add this to the file `.mvn/jvm.config` file. 
See: https://maven.apache.org/configure.html

I think this might be the better option, because not all plugins will obey the `argLine` property when starting a new process. Thoughts?

**Comment by jokowncode** [comment]
@bmhm Yes, but I don't know why such a change does not work when idea is executed, and only the VM parameter can be specified in the idea configuration. 
I may not have made it clear that this configuration should be placed in the configuration that loads maven-surefire-plugin, otherwise the `test `phase will report an error.

**Comment by jokowncode** [comment]
@bmhm Yeah, that makes sense. But for the time being, the only solution I can think of is the command line argument, and then I have to modify the configuration of the surefire-plugin of maven in pom.xml.

**Comment by bmhm** [comment]
@jokowncode then rather use `MAVEN_OPTS="-Duser.language=en"`

See: https://stackoverflow.com/a/12525359/1549977

There is also `JAVA_OPTS`

**Comment by jokowncode** [comment]
@bmhm Do not know why , IDEA does not apply MAVEN_OPTS when maven builds, but it is feasible to execute maven commands on the console. In other IDE, I haven't tested

**Comment by romani** [comment]
it is not first time when we make problems in non En locales.

the reason why we do not enforce local in maven for tests is because to check tests in all supported custom locales:
https://github.com/checkstyle/checkstyle/blob/53cddfd752c5a6b410a652ffb2bce2d9f6ff14ab/.travis.yml#L140-L145

but problems is Travis is not running `mvn verify` in Chinese, the most fragile part here is checkstyle part `mvn compile antrun:run@ant-phase-verify`. 

to resolve this issue we should update regexp in suppression file (I updated issue description with more details on this). 
But I would love to never come back to this problems again, so I would love if we be able:
1) reproduce problem in Travis, by MAVEN_OPTS, JAVA_OPTS, ..... any approach is ok (I remember tried this a while ago, but failed to reproduce problem). 
2) figure out why message is not the same as in Eng locale (I already tried to have to make it locale agnostic)
3) as 1) and 2) done, fix regexp in `suppression.xml`, let Travis validate this for future.

@jokowncode , are you ok to send PR that creates new travis build item to reproduce problem ?
please also investigate `2)`.

**Comment by jokowncode** [comment]
@romani Okay, but what do I have to do to interfere with Travis?

**Comment by romani** [comment]
Just update .travis.yml and put shell command in it

**Comment by srividyavaishnavi** [comment]
i am on it

**Comment by Ritik2604** [comment]
Why we are not discussing this?? I saw the code related to it that is all commented what happens? No doubt it is an important attribute that your project should be compilable or executable with language independent. @timurt @romani @jokowncode @bmarwell

**Comment by rnveach** [comment]
@Ritik2604 No one has been willing to put the time into fixing this. Admins are constantly busy reviewing and fixing dormant PRs. We rely on community help to resolve issues such as these.

**Comment by Ritik2604** [comment]
@rnveach By reading all the discussion on this topic and checking on attack overflow I concluded that we came up with a solution described by @romani but it needs some more discussion and changes and I think that solution might work with some modification I will love to work on this.

**Comment by romani** [comment]
Can we reproduce issue in CI ? I can not approve something that we can not verify that is become resolved after fix and CI can detect w somebody try to introduce this regression again in some PR.
We already accepted a lot of fix for Chinese locale, but people still have problems. So I try to understand where is a problem.
